openedx/core/djangoapps/course_groups/tests/test_cohorts.py

```diff
@@ -261,8 +267,8 @@ def test_get_cohort(self):
         )
 
     @ddt.data(
-        (True, 2),
-        (False, 6),
+        (True, 22),
+        (False, 32),
```

Comment:
Turns out, there were 2 separate problems here. Firstly, this test was not adding its user to a cohort properly in setup, so instead of "get an already assigned cohort" it became "calculate a default cohort and then assign to it", so it was an apples-to-oranges comparison.

Secondly, I decided not to make `get_cohort()` use `get_or_create()` logic, as the overhead of creating/preparing a default cohort on every `get()` had the potential to cause a performance hit on every single `get` request, which is much more common than the default assignment case.

The actual numbers that changed in this test are 3 and 9, rather than 2 and 6, accounting for the one extra step to get a `CourseUserGroup` after selecting a `CohortMembership`.